Use Julia's standard debug loggin mechanism instead of the custom verbose flag.

andreasnoack opened the pull request:
Minor thing but I think it's nicer to use the default logging machinery for this.

ararslan commented on docs/src/index.md:
````diff
@@ -256,3 +256,17 @@ of a ```LinPred``` type use a default step factor of 1.  The value of
+## Debugging failed fits
+In the rare cases when a fit of a generalized linear model fails, it can be useful
+to enable more output from the fitting steps. This can be done through
+the Julia logging mechanism by setting `ENV[JULIA_DEBUG] = GLM`. Enabling debug output
````
```suggestion
the Julia logging mechanism by setting `ENV["JULIA_DEBUG"] = GLM`. Enabling debug output
```
I can't recall whether the RHS also needs quotes, nor whether this needs to be specified before Julia and/or the package is loaded.

andreasnoack replied:
This version should be good.

ararslan reviewed (commented):
Looks good modulo one suggestion.

Do you think it would be worth doing a proper deprecation for the `verbose` keyword argument?

ararslan reviewed (approved):
LGTM, thanks! Could possibly be worthwhile to add a deprecation message but I defer to you on that.

andreasnoack commented:
@ararslan I've added a deprecation warning but I'm wondering if I should make it an `@info` instead of using `depwarn`. I think deprecation warnings have become mostly useless after they aren't enabled by default as nobody sees them anymore. 

I've also adjusted the debug messages to rely on the keyword functionality in the logging messages instead of interpolating the values into the string.

ararslan commented on src/glmfit.jl:
```diff
@@ -451,7 +450,10 @@ function StatsBase.fit!(m::AbstractGLM;
     end
-    if !issubset(keys(kwargs), (:maxIter, :minStepFac, :convTol))
+    if haskey(kwargs, :verbose)
+        Base.depwarn("""`verbose` argument is deprecated, use `ENV["JULIA_DEBUG"]=GLM` instead.""", :negbin)
```
I assume you mean
```suggestion
        Base.depwarn("""`verbose` argument is deprecated, use `ENV["JULIA_DEBUG"]=GLM` instead.""", :fit!)
```
?

andreasnoack replied:
indeed

ararslan commented on src/glmfit.jl:
```diff
@@ -490,7 +491,10 @@ function StatsBase.fit!(m::AbstractGLM,
     end
-    if !issubset(keys(kwargs), (:maxIter, :minStepFac, :convTol))
+    if haskey(kwargs, :verbose)
+        Base.depwarn("""`verbose` argument is deprecated, use `ENV["JULIA_DEBUG"]=GLM` instead.""", :negbin)
```
Here too?
```suggestion
        Base.depwarn("""`verbose` argument is deprecated, use `ENV["JULIA_DEBUG"]=GLM` instead.""", :fit!)
```

ararslan reviewed (commented):
The wrong function is getting associated with the deprecation in a couple of places but otherwise looks great! Good to merge once the suggested changes are incorporated.

ararslan commented:
> I've added a deprecation warning but I'm wondering if I should make it an `@info` instead of using `depwarn`. I think deprecation warnings have become mostly useless after they aren't enabled by default as nobody sees them anymore.

IMO it's best to use the functionality that's designed for this purpose, even if it means people are less likely to see it. It's a pet peeve of mine when deprecations are implemented in a way that bypasses the deprecation functionality because it can be difficult to silence when the caller isn't your own code.

> I've also adjusted the debug messages to rely on the keyword functionality in the logging messages instead of interpolating the values into the string.

Very nice!
